### PR TITLE
hypermedia calls can start directly from https://activities.api.prodev.d2l/my-unassessed-activities, don't have to go through root.

### PR DESCRIPTION
**Pull request by CarolBusman**
Also simplified some Promises to async/await for readability, added some null checks, rewrote `followLink()` to use `getHref()`.

### CODE REVIEW

File: demo/d2l-evaluation-hub/index.html [hunk 1]
```diff
@@ -34,7 +34,8 @@ <h3>Basic d2l-evaluation-hub demo</h3>
 			<h3>d2l-evaluation-hub-activities-list demo</h3>
 			<demo-snippet>
 				<template strip-whitespace>
-					<d2l-evaluation-hub-activities-list href="data/activities/activitiesList/root.json" token="whatever"></d2l-evaluation-hub-activities-list>
+					<d2l-evaluation-hub-activities-list href="data/activities/activitiesList/unassessedActivities.json" token="token"></d2l-evaluation-hub-activities-list>
+					<!-- <d2l-evaluation-hub-activities-list href="https://activities.api.proddev.d2l/my-unassessed-activities" token="token"></d2l-evaluation-hub-activities-list> -->
```
Comment (svanherk) [inline demo/d2l-evaluation-hub/index.html]: Do we need this comment here? Could we instead add a comment about this to the README, and use `<d2l-evaluation-hub>` instead?

File: test/d2l-evaluation-hub/data/root.json [hunk 2]
```diff
@@ -1,8 +0,0 @@
-{
```
Comment (svanherk) [inline test/d2l-evaluation-hub/data/root.json]: Is the test data the exact same as the demo data?  Should we instead combine them into the same place, or do we imagine them being different someday?

Comment (CarolBusman) [reply test/d2l-evaluation-hub/data/root.json]: The intent is to have the demo have more data to give a good example of what the table is going to look like, whereas the tests would have just enough data to perform the tests.

**Comment by svanherk** [comment]
FYI Jamie - For monolith work

**Comment by svanherk** [comment]
Corresponding LE PR: https://git.dev.d2l/projects/CORE/repos/le/pull-requests/13627/overview

**Comment by CarolBusman** [comment]
@CodeBaboon @pbrown-d2l Are there any concerns with this PR, or is it be approved?

**Pull request merged**